dex/elrond_dex_farm/src/lib.rs

```diff
@@ -341,21 +362,114 @@ pub trait Farm: rewards::RewardsModule + config::ConfigModule {
                 let locked_asset_factory_address = self.locked_asset_factory_address().get();
                 let result = self
                     .locked_asset_factory(locked_asset_factory_address)
-                    .create_and_forward(reward_amount.clone(), destination.clone())
+                    .create_and_forward(
+                        reward_amount.clone(),
+                        destination.clone(),
+                        opt_accept_funds_func.clone(),
+                    )
                     .execute_on_dest_context_custom_range(
-                        self.blockchain().get_gas_left() * 9 / 10,
+                        self.transfer_exec_gas_limit().get() * 2,
                         |_, after| (after - 1, after),
                     );
                 *reward_token_id = result.token_id;
                 *reward_nonce = result.token_nonce;
                 *reward_amount = result.amount;
             } else {
-                let _ = self.send().direct_esdt_via_transf_exec(
-                    destination,
-                    reward_token_id.as_esdt_identifier(),
+                self.send_tokens(
+                    reward_token_id,
+                    *reward_nonce,
                     reward_amount,
-                    &[],
-                );
+                    destination,
+                    opt_accept_funds_func,
+                )?;
+            }
+        }
+        Ok(())
+    }
+
+    fn send_tokens(
+        &self,
+        token: &TokenIdentifier,
+        nonce: Nonce,
+        amount: &Self::BigUint,
+        destination: &Address,
+        opt_accept_funds_func: &OptionalArg<BoxedBytes>,
+    ) -> SCResult<()> {
+        if nonce == 0 {
```

Comment:
I think this if is duplicated - previously you already checked with_locked_rewards - thus on else you will have unlocked rewards -> thus you can call send_fft - as before.